test/test_um2netcdf.py

```diff
@@ -651,6 +663,355 @@ def test_fix_units_do_nothing_no_um_units(ua_plev_cube):
+    """
+    Test that single point longitude and latitude coordinates
+    are provided with global bounds.
+    """
+
+    # Expected values after modification
+    expected_lat_bounds = um2nc.GLOBAL_COORD_BOUNDS[um2nc.LATITUDE]
+    expected_lon_bounds = um2nc.GLOBAL_COORD_BOUNDS[um2nc.LONGITUDE]
+
+    lat_coord_single = iris.coords.DimCoord(points=np.array([0]),
+                                            standard_name=um2nc.LATITUDE)
+    lon_coord_single = iris.coords.DimCoord(points=np.array([0]),
+                                            standard_name=um2nc.LONGITUDE)
+
+    cube_with_uv_coords = DummyCubeWithCoords(
+        dummy_cube=ua_plev_cube,
+        coords=[lat_coord_single, lon_coord_single]
+    )
+
+    assert not lat_coord_single.has_bounds()
+    assert not lon_coord_single.has_bounds()
+
+    um2nc.fix_latlon_coords(cube_with_uv_coords, um2nc.GRID_NEW_DYNAMICS,
+                            D_LAT_N96, D_LON_N96)
+
+    assert_has_bounds(lat_coord_single, lon_coord_single)
+    assert np.array_equal(lat_coord_single.bounds, expected_lat_bounds)
+    assert np.array_equal(lon_coord_single.bounds, expected_lon_bounds)
+
+
+def test_fix_latlon_coords_has_bounds(ua_plev_cube):
+    """
+    Test that existing coordinate bounds are not modified by
+    fix_latlon_coords.
+    """
+
+    # Expected values after modification
+    lon_bounds = np.array([[0, 1]])
+    lat_bounds = np.array([[10, 25]])
+
+    lat_coord = iris.coords.DimCoord(points=np.array([0]),
+                                     standard_name=um2nc.LATITUDE,
+                                     bounds=lat_bounds.copy())
+    lon_coord = iris.coords.DimCoord(points=np.array([0]),
+                                     standard_name=um2nc.LONGITUDE,
+                                     bounds=lon_bounds.copy())
+
+    cube_with_uv_coords = DummyCubeWithCoords(
+        dummy_cube=ua_plev_cube,
+        coords=[lat_coord, lon_coord]
+    )
+    assert_has_bounds(lat_coord, lon_coord)
+
+    um2nc.fix_latlon_coords(cube_with_uv_coords, um2nc.GRID_NEW_DYNAMICS,
+                            D_LAT_N96, D_LON_N96)
+
+    assert np.array_equal(lat_coord.bounds, lat_bounds)
+    assert np.array_equal(lon_coord.bounds, lon_bounds)
+
+
+def test_fix_latlon_coords_missing_coord_error(ua_plev_cube):
+    """
+    Test that fix_latlon_coords raises the right type of error when a cube
+    is missing coordinates.
+    """
+    def _raise_CoordinateNotFoundError(coord_name):
+        # Include an argument "coord_name" to mimic the signature of an
+        # Iris cube's ".coord" method
+        raise iris.exceptions.CoordinateNotFoundError(coord_name)
+
+    # Replace coord method to raise UnsupportedTimeSeriesError
+    ua_plev_cube.coord = _raise_CoordinateNotFoundError
+
+    with (
+        pytest.raises(um2nc.UnsupportedTimeSeriesError)
+    ):
+        um2nc.fix_latlon_coords(ua_plev_cube, um2nc.GRID_NEW_DYNAMICS,
+                                D_LAT_N96, D_LON_N96)
```

Comment:
This test can be simplified once the `CoordinateNotFoundError` fix from the earlier suggestion:
https://github.com/ACCESS-NRI/um2nc-standalone/pull/63/files#r1749685839

I think simply passing in a dummy cube should cause the required breakage, if the `lat/lon` coords are missing. There might be a duplicate of this test in another PR. I can't recall where, but it might allow this test to be deleted.

Reply:
The test definitely becomes simpler after adding the `CoordinateNotFoundError`. Added in 6a4c876